src/movies.js

```diff
@@ -2,14 +2,122 @@
 
 // Iteration 1: Ordering by year - Order by year, ascending (in growing order)
 
+function orderByYear(newArr) {
+  let array2 = [...newArr];
+  let orderByYear = array2.sort(function(objA, objB) {
+    if (objA.year < objB.year) {
+      return -1;
+    }
+    if (objA.year > objB.year) {
+      return 1;
+    }
+    if (objA.year === objB.year) {
+      if (objA.title < objB.title) {
+        return -1;
+      }
+      if (objA.title > objB.title) {
+        return 1;
+      }
+      if (objA.title === objB.title) {
+        return 0;
+      }
+    }
+  });
+  return orderByYear;
+}
+
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct
 
+/*Go ahead and create a howManyMovies() function that receives an array as a parameter and filter 👀 
+the array so we can have only the drama movies where Steven Spielberg is the director.
+*/
+
+function howManyMovies(array) {
+  /* if (num=0){
+
+    };
+    return num
+    */
+  if (array.length === 0) {
+    return 0;
+  }
+  const newArray = array.filter(function(obj, i, originalArr) {
+    if (obj.director === "Steven Spielberg" && obj.genre.includes("Drama")) {
+      return true;
+    }
+  });
+
+  return newArray.length;
+}
+
 // Iteration 3: Alphabetic Order - Order by title and print the first 20 titles
 
+function orderAlphabetically(newArray) {
+  let array3 = [...newArray];
+  let order = array3
+    .sort(function(objA, objB) {
+      if (objA.title < objB.title) {
+        return -1;
+      }
+      if (objA.title > objB.title) {
+        return 1;
+      }
+      if (objA.title === objB.title) {
+        return 0;
+      }
+    })
+    .map(function(movie, index, originalArr) {
+      return movie.title;
+    });
+  if (order.length > 20) {
+    return order.slice(0, 20);
+  }
+  return order;
+}
+
 // Iteration 4: All rates average - Get the average of all rates with 2 decimals
 
+function ratesAverage(newArr) {
```

Comment:
You had good logic on this function but there were a couple issues:

- the Jasmine check for movies between 8 and 8.9 is something done automatically by jasmine, you in the function you were supposed to just get all the rates.
- Because you are applying .reduce directly to the .map you are working with the list of rates. Because of that, inside the .reduce you are not passing an object but a rate number. So instead of doing rateObj.rate you would just do rateObj. Naming of parameters probably confused you here.
- Remember .reduce applies the changes to the variable being declared ( rates ), so to get the average in the variable average you should use the variable rates and not the variable updatedAcc. The variable updatedAcc has a scope inside the .reduce method and cannot be accesed from outside.
- Finally, the .toFixed method changes the number to a string, so you will hace to wrap everything into a parseFloar( ) to change it back to a number.